src/main/java/com/algorand/algosdk/crypto/Address.java

```diff
@@ -198,4 +202,20 @@ public boolean compareTo(String address) {
         }
         return this.toString().equals(address);
     }
+
+    /**
+     * Get the escrow address of an application.
+     * @param appID The ID of the application.
+     * @return The address corresponding to that application's escrow account.
+     * @throws NoSuchAlgorithmException
+     */
+    public static Address forApplication(long appID) throws NoSuchAlgorithmException {
+        byte[] encodedID = Uint64Encoder.encode(appID);
+
+        byte[] toHash = new byte[encodedID.length + APP_ID_PREFIX.length];
+        System.arraycopy(APP_ID_PREFIX, 0, toHash, 0, APP_ID_PREFIX.length);
+        System.arraycopy(encodedID, 0, toHash, APP_ID_PREFIX.length, encodedID.length);
```

Comment:
Please use ByteArrayOutputStream instead of arraycopy:
https://github.com/algorand/java-algorand-sdk/blob/develop/src/main/java/com/algorand/algosdk/account/Account.java#L542